go/s3/s3SetTls12.go

```diff
@@ -0,0 +1,125 @@
+// snippet-comment:[The snippet tags are for the AWS SDK for Go Developer Guide. Do not remove.]
+/*
+   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+   This file is licensed under the Apache License, Version 2.0 (the "License").
+   You may not use this file except in compliance with the License. A copy of
+   the License is located at
+
+    http://aws.amazon.com/apache2.0/
+
+   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+   specific language governing permissions and limitations under the License.
+*/
+// snippet-start:[s3.go.set_tls_12]
+package main
+
+import (
+    "crypto/tls"
+    "flag"
+    "fmt"
+    "net/http"
+
+    "github.com/aws/aws-sdk-go/aws"
+    "github.com/aws/aws-sdk-go/aws/session"
+    "github.com/aws/aws-sdk-go/service/s3"
+    "golang.org/x/net/http2"
+)
+
+// GetTLSVersion gets the version of TLS you have configured
+// snippet-start:[s3.go.get_tls_version]
+func GetTLSVersion(tr *http.Transport) string {
+    switch tr.TLSClientConfig.MinVersion {
+    case tls.VersionTLS10:
+        return "TLS 1.0"
+    case tls.VersionTLS11:
+        return "TLS 1.1"
+    case tls.VersionTLS12:
+        return "TLS 1.2"
+    case tls.VersionTLS13:
+        return "TLS 1.3"
+    }
+
+    return "Unknown"
+}
```

Comment:
Should we also highlight how to retrieve `*http.transport` from a s3client?

Reply:
Could you use that to determine your default TLS version? If so, I'd love to add the code.

Reply:
I meant, the GetTLSVersion takes in *http.transport as a parameter.(I our case tr is that parameter). Customers might find it helpful to know how to get that `tr` from a client.

One way is like below: 
```
if tr, ok := s3Client.Config.HTTPClient.Transport.(*http.Transport); ok {
   log.Printf("Client uses %v", GetTLSVersion(tr))
}
```

Reply:
Gotcha.